presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -594,6 +595,8 @@ private PlanWithProperties planTableScan(TableScanNode node, Expression predicat
                             .map(node.getAssignments()::get)
                             .collect(toImmutableSet())));
 
+            layouts = filteredCopy(layouts, layout -> !layout.getLayout().getPredicate().isNone());
```

Comment:
Instead of this, I'd rather put the following right after `checkState(!layouts.isEmpty(), "No usable layouts);`:
```
if (layouts.stream().anyMatch( layout -> layout.getLayout().getPredicate().isNone()) ) {
    return Values();
}
```